packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -32,14 +33,20 @@ export const Appender = forwardRef(
 			};
 		}, [] );
 
+		const blockTitle = useBlockDisplayTitle( {
+			clientId,
```

Comment:
There are a couple of things to be careful about with this if/when the feature is ported to the main List View.

This clientId is for the selected block, whereas it should be for the parent block. For the main List View it'd always append at root level, so it'd have to say 'Append to post' or something.